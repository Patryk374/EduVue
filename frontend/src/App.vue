<template>
  <h1>Kursy</h1>
  <ul>
    <li v-for="c in courses" :key="c.id">
      <strong>{{ c.nazwa }}</strong> - {{ c.dlugosc_trwania }} - {{ c.sugerowany_przedzial_wiekowy }}<br />
      {{ c.szczegoly }}
    </li>
  </ul>
</template>

<script setup>
import { ref, onMounted } from 'vue'

const courses = ref([])

onMounted(async () => {
  const res = await fetch('/courses')
  courses.value = await res.json()
})
</script>
