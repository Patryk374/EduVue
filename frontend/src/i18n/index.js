import { createI18n } from 'vue-i18n'

const messages = {
  en: {
    nav: {
      home: 'Home',
      courses: 'Courses',
      news: 'News'
    },
    home: {
      title: 'Learn Programming with EduVue',
      subtitle: 'Master programming through interactive courses and hands-on projects',
      description: 'Start your learning journey today with our comprehensive courses',
      popularCourses: 'Popular Courses',
      startLearning: 'Start Learning',
      whyChoose: 'Why Choose EduVue?',
      features: {
        interactive: {
          title: 'Interactive Learning',
          description: 'Learn by doing with hands-on projects and real-world examples'
        },
        experts: {
          title: 'Expert Instructors',
          description: 'Learn from industry professionals with years of experience'
        },
        flexible: {
          title: 'Flexible Schedule',
          description: 'Learn at your own pace with 24/7 access to course materials'
        }
      }
    },
    courses: {
      title: 'Available Courses',
      subtitle: 'Find the perfect course to start your programming journey',
      search: 'Search courses...',
      filter: 'Filter by category',
      noCourses: 'No courses found',
      backToCourses: 'Back to Courses',
      returnToHome: 'Return to Home',
      clearFilters: 'Clear filters',
      categories: {
        programming: 'Programming',
        web: 'Web Development'
      },
      levels: {
        beginner: 'Beginner',
        intermediate: 'Intermediate',
        advanced: 'Advanced'
      }
    },
    course: {
      start: 'Start Course',
      review: 'Review',
      progress: 'Progress',
      lessons: 'Lessons',
      duration: 'Duration',
      level: 'Level',
      completedLessons: 'Completed Lessons',
      remainingTime: 'Remaining Time',
      details: 'Course Details'
    },
    footer: {
      rights: 'All rights reserved'
    },
    news: {
      title: 'News from the programming world',
      subtitle: 'Discover the latest articles and trends via Hacker News',
      loading: 'Loading news...'
    }
  },
  pl: {
    nav: {
      home: 'Strona główna',
      courses: 'Kursy',
      news: 'Nowinki'
      
    },
    home: {
      title: 'Ucz się programowania z EduVue',
      subtitle: 'Opanuj programowanie dzięki interaktywnym kursom i praktycznym projektom',
      description: 'Rozpocznij swoją przygodę z nauką już dziś z naszymi kompleksowymi kursami',
      popularCourses: 'Popularne kursy',
      startLearning: 'Rozpocznij naukę',
      whyChoose: 'Dlaczego EduVue?',
      features: {
        interactive: {
          title: 'Interaktywna Nauka',
          description: 'Ucz się poprzez praktykę z projektami i przykładami z życia wziętymi'
        },
        experts: {
          title: 'Eksperci',
          description: 'Ucz się od profesjonalistów z wieloletnim doświadczeniem'
        },
        flexible: {
          title: 'Elastyczny Harmonogram',
          description: 'Ucz się we własnym tempie z dostępem do materiałów 24/7'
        }
      }
    },
    courses: {
      title: 'Dostępne kursy',
      subtitle: 'Znajdź idealny kurs, aby rozpocząć swoją przygodę z programowaniem',
      search: 'Szukaj kursów...',
      filter: 'Filtruj według kategorii',
      noCourses: 'Nie znaleziono kursów',
      backToCourses: 'Powrót do kursów',
      returnToHome: 'Powrót do strony głównej',
      clearFilters: 'Wyczyść filtry',
      categories: {
        programming: 'Programowanie',
        web: 'Tworzenie stron WWW'
      },
      levels: {
        beginner: 'Początkujący',
        intermediate: 'Średniozaawansowany',
        advanced: 'Zaawansowany'
      }
    },
    course: {
      start: 'Rozpocznij kurs',
      review: 'Powtórz',
      progress: 'Postęp',
      lessons: 'Lekcje',
      duration: 'Czas trwania',
      level: 'Poziom',
      completedLessons: 'Ukończone lekcje',
      remainingTime: 'Pozostały czas',
      details: 'Szczegóły kursu'
    },
    footer: {
      rights: 'Wszelkie prawa zastrzeżone'
    },
    news: {
      title: 'Nowinki ze świata programowania',
      subtitle: 'Poznaj najnowsze artykuły i trendy dzięki Hacker News',
      loading: 'Ładowanie nowinek...'
    }
  }
}

const i18n = createI18n({
  legacy: false,
  locale: 'pl',
  fallbackLocale: 'en',
  messages
})

export default i18n 