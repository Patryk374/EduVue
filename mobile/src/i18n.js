import { I18n } from 'i18n-js';
import * as Localization from 'expo-localization';

// English and Polish translations
const en = {
  nav: { home: 'Home', courses: 'Courses' },
  home: {
    title: 'Learn Programming with EduVue',
    subtitle: 'Master programming through interactive courses',
    startLearning: 'Start Learning'
  },
  courses: {
    title: 'Available Courses',
    search: 'Search courses...',
    enroll: 'Enroll',
    returnToHome: 'Return to Home'
  },
  toggleTheme: 'Toggle Theme',
  menu: {
    dark: 'Dark Theme',
    light: 'Light Theme',
    language: 'Change Language'
  }
};

const pl = {
  nav: { home: 'Strona główna', courses: 'Kursy' },
  home: {
    title: 'Ucz się programowania z EduVue',
    subtitle: 'Opanuj programowanie dzięki interaktywnym kursom',
    startLearning: 'Rozpocznij naukę'
  },
  courses: {
    title: 'Dostępne kursy',
    search: 'Szukaj kursów...',
    enroll: 'Zapisz się',
    returnToHome: 'Powrót do strony głównej'
  },
  toggleTheme: 'Przełącz motyw',
  menu: {
    dark: 'Ciemny motyw',
    light: 'Jasny motyw',
    language: 'Zmień język'
  }
};

const i18n = new I18n({ en, pl });
i18n.enableFallback = true;
i18n.locale = Localization.locale || 'en';

export default i18n;
