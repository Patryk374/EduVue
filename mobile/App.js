import React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import HomeScreen from './src/screens/HomeScreen';
import CoursesScreen from './src/screens/CoursesScreen';
import CourseDetailScreen from './src/screens/CourseDetailScreen';
import { ThemeProvider } from './src/ThemeContext';
import { LanguageProvider } from './src/LanguageContext';
import i18n from './src/i18n';

const Stack = createNativeStackNavigator();

export default function App() {
  return (
    <ThemeProvider>
      <LanguageProvider>
        <NavigationContainer>
          <Stack.Navigator>
            <Stack.Screen name="Home" component={HomeScreen} options={{ title: i18n.t('nav.home') }} />
            <Stack.Screen name="Courses" component={CoursesScreen} options={{ title: i18n.t('nav.courses') }} />
            <Stack.Screen name="CourseDetail" component={CourseDetailScreen} options={{ title: 'Course' }} />
          </Stack.Navigator>
        </NavigationContainer>
      </LanguageProvider>
    </ThemeProvider>
  );
}