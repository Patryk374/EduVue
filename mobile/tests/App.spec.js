import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import renderer, { act as rendererAct } from 'react-test-renderer';
import App from '../App';
import i18n from '../src/i18n';
 
// Test 1: Renderowanie głównego komponentu App
it('renders NavigationContainer with stack screens and i18n titles', () => {
  let tree;
  rendererAct(() => {
    tree = renderer.create(<App />);
  });
  const screens = tree.root.findAll(node => node.props && node.props.name);
  const names = screens.map(s => s.props.name);
  const uniqueNames = [...new Set(names)];
  expect(uniqueNames).toEqual(['Home', 'Courses', 'CourseDetail']);
  const homeScreen = screens.find(s => s.props.name === 'Home');
  const coursesScreen = screens.find(s => s.props.name === 'Courses');
  expect(homeScreen.props.options.title).toBe(i18n.t('nav.home'));
  expect(coursesScreen.props.options.title).toBe(i18n.t('nav.courses'));
});
 
// Test 9: Komponent nawigacji – routing działa poprawnie
it('navigates to Courses screen from Home', async () => {

  const { getByText, findByPlaceholderText } = render(<App />);
  fireEvent.press(getByText(i18n.t('home.startLearning')));
  const search = await findByPlaceholderText(i18n.t('courses.search'));
  expect(search).toBeTruthy();
});
 
// Test 10: Test snapshotowy App
it('matches App snapshot', () => {
  let tree;
  rendererAct(() => {
    tree = renderer.create(<App />);
  });
  const json = tree.toJSON();
  expect(json).toMatchSnapshot();
});
