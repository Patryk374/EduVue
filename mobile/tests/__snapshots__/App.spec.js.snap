// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches App snapshot 1`] = `
<View>
  <View>
    <View>
      <View
        name="Home"
        options={
          {
            "title": "Home",
          }
        }
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
                "padding": 20,
              },
              {
                "backgroundColor": undefined,
              },
            ]
          }
        >
          <Text
            style={
              [
                {
                  "fontSize": 24,
                  "fontWeight": "bold",
                  "marginBottom": 16,
                  "textAlign": "center",
                },
                {
                  "color": undefined,
                },
              ]
            }
          >
            Learn Programming with EduVue
          </Text>
          <Text
            style={
              [
                {
                  "fontSize": 16,
                  "marginBottom": 24,
                  "textAlign": "center",
                },
                {
                  "color": undefined,
                },
              ]
            }
          >
            Master programming through interactive courses
          </Text>
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                [
                  {},
                ]
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#007AFF",
                      "fontSize": 18,
                      "margin": 8,
                      "textAlign": "center",
                    },
                  ]
                }
              >
                Start Learning
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        name="Courses"
        options={
          {
            "title": "Courses",
          }
        }
      >
        <View
          style={
            [
              {
                "flex": 1,
                "padding": 16,
              },
              {
                "backgroundColor": undefined,
              },
            ]
          }
        >
          <TextInput
            mode="outlined"
            onChangeText={[Function]}
            placeholder="Search courses..."
            style={
              {
                "borderRadius": 4,
                "marginBottom": 12,
                "padding": 8,
              }
            }
            value=""
          />
          <RCTScrollView
            data={
              [
                {
                  "description": "Podstawy programowania dla początkujących",
                  "description_en": "Basics of programming for beginners",
                  "duration": "4 tygodnie",
                  "id": "1",
                  "level": "beginner",
                  "title": "Wprowadzenie do programowania",
                  "title_en": "Introduction to Programming",
                },
                {
                  "description": "Poznaj zaawansowane koncepcje JavaScript",
                  "description_en": "Learn advanced JavaScript concepts",
                  "duration": "6 tygodni",
                  "id": "2",
                  "level": "advanced",
                  "title": "Zaawansowany JavaScript",
                  "title_en": "Advanced JavaScript",
                },
                {
                  "description": "Tworzenie atrakcyjnych i responsywnych stron",
                  "description_en": "Creating attractive and responsive websites",
                  "duration": "5 tygodni",
                  "id": "3",
                  "level": "beginner",
                  "title": "Projektowanie stron WWW",
                  "title_en": "Web Design",
                },
              ]
            }
            getItem={[Function]}
            getItemCount={[Function]}
            keyExtractor={[Function]}
            onContentSizeChange={[Function]}
            onLayout={[Function]}
            onMomentumScrollBegin={[Function]}
            onMomentumScrollEnd={[Function]}
            onScroll={[Function]}
            onScrollBeginDrag={[Function]}
            onScrollEndDrag={[Function]}
            removeClippedSubviews={false}
            renderItem={[Function]}
            scrollEventThrottle={0.0001}
            stickyHeaderIndices={[]}
            viewabilityConfigCallbackPairs={[]}
          >
            <View>
              <View
                onFocusCapture={[Function]}
                onLayout={[Function]}
                style={null}
              >
                <View
                  onPress={[Function]}
                  style={
                    [
                      {
                        "marginBottom": 12,
                      },
                      {
                        "backgroundColor": "#fff",
                      },
                    ]
                  }
                >
                  <View
                    source={
                      {
                        "testUri": "../../../assets/icon.png",
                      }
                    }
                    style={
                      {
                        "height": 120,
                      }
                    }
                  />
                  <View
                    left={[Function]}
                  >
                    <Text>
                      Introduction to Programming
                    </Text>
                  </View>
                  <View>
                    <Text
                      style={
                        {
                          "color": "#555",
                        }
                      }
                      variant="bodyMedium"
                    >
                      Basics of programming for beginners
                    </Text>
                  </View>
                  <View>
                    <View
                      accessibilityRole="button"
                      accessibilityState={
                        {
                          "busy": undefined,
                          "checked": undefined,
                          "disabled": undefined,
                          "expanded": undefined,
                          "selected": undefined,
                        }
                      }
                      accessibilityValue={
                        {
                          "max": undefined,
                          "min": undefined,
                          "now": undefined,
                          "text": undefined,
                        }
                      }
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        {
                          "opacity": 1,
                        }
                      }
                    >
                      <View
                        style={
                          [
                            {},
                          ]
                        }
                      >
                        <Text
                          style={
                            [
                              {
                                "color": "#007AFF",
                                "fontSize": 18,
                                "margin": 8,
                                "textAlign": "center",
                              },
                            ]
                          }
                        >
                          Enroll
                        </Text>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
              <View
                onFocusCapture={[Function]}
                onLayout={[Function]}
                style={null}
              >
                <View
                  onPress={[Function]}
                  style={
                    [
                      {
                        "marginBottom": 12,
                      },
                      {
                        "backgroundColor": "#fff",
                      },
                    ]
                  }
                >
                  <View
                    source={
                      {
                        "testUri": "../../../assets/icon.png",
                      }
                    }
                    style={
                      {
                        "height": 120,
                      }
                    }
                  />
                  <View
                    left={[Function]}
                  >
                    <Text>
                      Advanced JavaScript
                    </Text>
                  </View>
                  <View>
                    <Text
                      style={
                        {
                          "color": "#555",
                        }
                      }
                      variant="bodyMedium"
                    >
                      Learn advanced JavaScript concepts
                    </Text>
                  </View>
                  <View>
                    <View
                      accessibilityRole="button"
                      accessibilityState={
                        {
                          "busy": undefined,
                          "checked": undefined,
                          "disabled": undefined,
                          "expanded": undefined,
                          "selected": undefined,
                        }
                      }
                      accessibilityValue={
                        {
                          "max": undefined,
                          "min": undefined,
                          "now": undefined,
                          "text": undefined,
                        }
                      }
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        {
                          "opacity": 1,
                        }
                      }
                    >
                      <View
                        style={
                          [
                            {},
                          ]
                        }
                      >
                        <Text
                          style={
                            [
                              {
                                "color": "#007AFF",
                                "fontSize": 18,
                                "margin": 8,
                                "textAlign": "center",
                              },
                            ]
                          }
                        >
                          Enroll
                        </Text>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
              <View
                onFocusCapture={[Function]}
                onLayout={[Function]}
                style={null}
              >
                <View
                  onPress={[Function]}
                  style={
                    [
                      {
                        "marginBottom": 12,
                      },
                      {
                        "backgroundColor": "#fff",
                      },
                    ]
                  }
                >
                  <View
                    source={
                      {
                        "testUri": "../../../assets/icon.png",
                      }
                    }
                    style={
                      {
                        "height": 120,
                      }
                    }
                  />
                  <View
                    left={[Function]}
                  >
                    <Text>
                      Web Design
                    </Text>
                  </View>
                  <View>
                    <Text
                      style={
                        {
                          "color": "#555",
                        }
                      }
                      variant="bodyMedium"
                    >
                      Creating attractive and responsive websites
                    </Text>
                  </View>
                  <View>
                    <View
                      accessibilityRole="button"
                      accessibilityState={
                        {
                          "busy": undefined,
                          "checked": undefined,
                          "disabled": undefined,
                          "expanded": undefined,
                          "selected": undefined,
                        }
                      }
                      accessibilityValue={
                        {
                          "max": undefined,
                          "min": undefined,
                          "now": undefined,
                          "text": undefined,
                        }
                      }
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        {
                          "opacity": 1,
                        }
                      }
                    >
                      <View
                        style={
                          [
                            {},
                          ]
                        }
                      >
                        <Text
                          style={
                            [
                              {
                                "color": "#007AFF",
                                "fontSize": 18,
                                "margin": 8,
                                "textAlign": "center",
                              },
                            ]
                          }
                        >
                          Enroll
                        </Text>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </RCTScrollView>
        </View>
      </View>
      <View
        name="CourseDetail"
        options={
          {
            "title": "Course",
          }
        }
      />
    </View>
  </View>
</View>
`;
